src/repositories/users.py

```diff
@@ -1,24 +1,49 @@
 # ------- REPOSITORY FILE -------
-from ..models import UserBase, UserCreate
+from ..models import UserCreate, UserResponse, UserWithRoleResponse
 
 
 class UsersRepository:
     def __init__(self, db_session):
         self.db_session = db_session
 
-    async def get_user_by_email(self, email: str) -> UserBase:
+    # Get users
+
+    async def get_user_by_email(self, email: str) -> UserResponse:
         async with self.db_session.transaction():
-            query = "SELECT * FROM d_roles.users WHERE email = :email"
+            query = """
+            SELECT * FROM d_roles.users WHERE U.email = :email
+            """
             return await self.db_session.fetch_one(query, {"email": email})
 
-    async def add_user(self, user: UserCreate) -> int:
+    async def get_user_by_id(self, user_id: int) -> UserResponse:
+        async with self.db_session.transaction():
+            query = """
+            SELECT * FROM d_roles.users as U WHERE U.id = :id
+            """
+            return await self.db_session.fetch_one(query, {"id": user_id})
+
+    async def get_users_by_group_id(self, group_id: int) -> list[UserWithRoleResponse]:
         async with self.db_session.transaction():
-            query = "INSERT INTO d_roles.users (email, is_email_confirmed) VALUES (:email, :is_email_confirmed) RETURNING id"
+            query = """
+                SELECT U.email, U.sub_pro_connect, U.created_at, R.role_name, R.is_admin
+                FROM d_roles.users as U
+                INNER JOIN d_roles.group_user_relations as TUR ON TUR.group_id = :group_id
+                INNER JOIN d_roles.roles as R ON TUR.role_id = R.id
```

Comment:
```suggestion
                INNER JOIN d_roles.group_user_relations as TUR ON TUR.user_id = U.id
                INNER JOIN d_roles.roles as R ON TUR.role_id = R.id
                WHERE TUR.group_id = :group_id
```
Currently doing a cross join between `users` and `group_user_relations`.